### PR TITLE
Sort post, Infinite scroll and some minor changes

### PR DESCRIPTION
**Pull request by ravikumar0403**
### Changelog 
- Added document title for each page
- Users can now sort posts based on oldest, lastest, and trending
- Users can now share posts with any user by copying the link
- Added Infinite Scroll on Feed and Explore page
- Added support for emoji while creating a post

### CODE REVIEW

**Review by yashpatel1797 — APPROVED** [review]
code looks clean
file structure is good
design looks consistent

File: public/index.html [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-    />
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
-    <!--
+
+<head>
+  <meta charset="utf-8" />
+  <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta name="theme-color" content="#000000" />
+  <meta name="description" content="Web site created using create-react-app" />
```
Comment (parthpandyappp) [inline public/index.html]: Can describe about your application in this meta tag.

**Review by krgarima — APPROVED** [review]
Good work. Everything is working smoothly. Nice implementation of logic. Good to merge.

File: src/components/PostCard.js [hunk 2]
```diff
@@ -49,13 +56,13 @@ export const PostCard = ({ post }) => {
...
-              className="p-2 rounded-full text-2xl hover:bg-teal-50 hover:text-teal-500"
+              className="p-2 mr-1 rounded-full text-2xl hover:bg-teal-50 hover:text-teal-500"
               title="like"
               onClick={() =>
-                dispatch(likePost({ post_id: post._id, user_id: user._id }))
+                dispatch(likePost({ post_id: post._id, user_id: user?._id }))
               }
             >
-              {post.likes.includes(user._id) ? (
+              {post.likes.includes(user?._id) ? (
```
Comment (parthpandyappp) [inline src/components/PostCard.js:65]: You can apply this conditional in the `className`. If `user?.id` then apply your classes else not.

**Review by parthpandyappp — APPROVED** [review]
Looks good to me. Left some comments :+1: 
Go ahead with the merge :grinning:

File: public/index.html (hunk 1 above)
Comment (sashaboi) [inline public/index.html]: Description helps with creating preview , maybe you could add relevant desc.

File: src/App.js [hunk 3]
```diff
@@ -9,9 +9,11 @@ import jwtDecode from "jwt-decode";
 import axios from "axios";
 import { useScrollToTop } from "hooks/useScrollToTop";
 import { USER } from "utils/constants";
+import { useDocumentTitle } from "hooks/useDocumentTitle";
 
 const App = () => {
   useScrollToTop();
```
Comment (sashaboi) [inline src/App.js:15]: love this functionality

File: src/components/PostInputModal.js [hunk 4]
```diff
@@ -68,18 +75,37 @@ export const PostInputModal = () => {
...
+                className={`${
+                  showEmojiPicker ? "" : "hidden"
+                } absolute emoji-container`}
+              >
+                <EmojiPicker onEmojiClick={onEmojiClick} />
+              </div>
+            </div>
+            {!selectedPost && (
+              <label className="p-2 flex items-center mr-2 rounded-full block hover:bg-teal-100 hover:text-teal-600">
+                <RiImageAddLine size={"1.2rem"} />
```
Comment (sashaboi) [inline src/components/PostInputModal.js:99]: Could have a common classname for this line ?

File: src/components/SortPost.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+          Sort by: {formatSortBy(sortBy)}
+          <AiOutlineCaretDown className="mx-2" />
+        </button>
+        {showDropdown && (
+          <div className="absolute right-0 flex flex-col mx-2 bg-slate-50 dark:bg-slate-600 p-2 rounded min-w-max z-10">
+            <button
+              className="py-1 px-8 rounded hover:text-teal-500 hover:bg-teal-50"
+              onClick={() => handleClick(TRENDING)}
+            >
+              Trending
```
Comment (sashaboi) [inline src/components/SortPost.js:41]: You could add a quick small icon here along with each option to increase UX

File: src/utils/sortPosts.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import { LATEST_FIRST, OLDEST_FIRST, TRENDING } from "./constants";
+
+export const sortPosts = (posts, sortBy = LATEST_FIRST) => {
+  if (posts.length === 0) return [];
+  switch (sortBy) {
+    case LATEST_FIRST:
+      return posts;
+    case OLDEST_FIRST:
+      return [...posts].sort((a, b) =>
+        dayjs(a.createdAt).diff(dayjs(b.createdAt))
```
Comment (sashaboi) [inline src/utils/sortPosts.js:11]: Loved how this works

**Review by sashaboi — APPROVED** [review]
Amazing work on this project ! 
made a few suggestions and comments for future versions..

**Pull request merged**